### PR TITLE
[runtime env] plugin refactor[3/n]: support strong type by @dataclass

### PR DESCRIPTION
**Pull request by SongGuyang**
## Why are these changes needed?

This PR is blocked by https://github.com/ray-project/ray/pull/26255.

The related prototype https://github.com/ray-project/ray/pull/25387

### CODE REVIEW

File: python/ray/tests/test_runtime_env_strong_type.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+from typing import List
+from ray.runtime_env import RuntimeEnv
+from ray.runtime_env.types.pip import Pip
+from dataclasses import dataclass
+
+
+@dataclass
+class ValueType:
+    nfeild1: List[str]
```
Comment (architkulkarni) [inline python/ray/tests/test_runtime_env_strong_type.py:14]: `feild` -> `field` throughout

File: python/ray/tests/test_runtime_env_strong_type.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    assert test_plugin_2.feild1[0].nfeild1 == ["a", "b", "c"]
+    assert test_plugin_2.feild1[0].nfeild2 is False
+    assert test_plugin_2.feild1[1].nfeild1 == ["d", "e"]
+    assert test_plugin_2.feild1[1].nfeild2 is True
+    assert test_plugin_2.feild2 == "abc"
+
+
+@pytest.mark.skipif(
+    sys.platform == "win32",
+    reason="Pip option not supported on Windows.",
```
Comment (architkulkarni) [inline python/ray/tests/test_runtime_env_strong_type.py:48]: It should be supported now, can we enable tests on Windows in this PR and in the future?  (If it fails and is blocking, we can consider it on a case-by-case basis)

**Review by architkulkarni — APPROVED** [review]
Thanks for adding this and for adding the tests!  Just some very minor comments.

One thing we need to do is to update our `LICENSE` file to include `dacite`.  Could you please do this before merging? There should be examples there to go off of, but let me know if you have any questions.

**Comment by SongGuyang** [comment]
@architkulkarni  license added [here](https://github.com/ray-project/ray/pull/26296/files#diff-c693279643b8cd5d248172d9c22cb7cf4ed163a3c98c8a3f69c2717edd3eacb7R406). Thanks for your guide!

**Review by raulchen — APPROVED** [review]

**Pull request merged**